component/loki/source/kafka/kafka.go

```diff
@@ -26,6 +27,16 @@ func init() {
 	})
 }
 
+// Takes in a variable of type `rivertypes.Secret`, converts it to an OptionalSecret in order to extract the string value
+func ConvertSecretToString(secret rivertypes.Secret) (string, error) {
```

Comment:
I think this function should be unexported. Also, consider moving into `rivertypes` package if could be used in other components.

Reply:
Yes, it can be used in other components as well as we move to `Secret` type for all sensitive data. I'll move it into the `rivertypes` package